src/LibraryManager/Providers/Unpkg/NpmPackageInfoFactory.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using Newtonsoft.Json.Linq;
+
+#if NET472
+using System.ComponentModel.Composition;
+#endif
+
+namespace Microsoft.Web.LibraryManager.Providers.Unpkg
+{
+#if NET472
+    [Export(typeof(INpmPackageInfoFactory))]
+#endif
+    internal sealed class NpmPackageInfoFactory : INpmPackageInfoFactory
+    {
+        public const string NpmPackageInfoUrl = "https://registry.npmjs.org/{0}";
+        public const string NpmLatestPackageInfoUrl = "https://registry.npmjs.org/{0}/latest";
+
+        private readonly Dictionary<string, NpmPackageInfo> _cachedPackages = new Dictionary<string, NpmPackageInfo>();
+
+        public async Task<NpmPackageInfo> GetPackageInfoAsync(string packageName, CancellationToken cancellationToken)
+        {
+            if (!_cachedPackages.TryGetValue(packageName, out NpmPackageInfo packageInfo))
+            {
+                packageInfo = await GetNewPackageInfoAsync(packageName, cancellationToken).ConfigureAwait(false);
+
+                if (packageInfo != null)
+                {
+                    _cachedPackages[packageName] = packageInfo;
+                }
+            }
+
+            return packageInfo;
+        }
+
+        private async Task<NpmPackageInfo> GetNewPackageInfoAsync(string packageName, CancellationToken cancellationToken)
+        {
+            NpmPackageInfo packageInfo = null;
+
+            if (packageName.StartsWith("@", StringComparison.Ordinal))
+            {
+                packageInfo = await GetPackageInfoForScopedPackageAsync(packageName, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                packageInfo = await GetPackageInfoForUnscopedPackageAsync(packageName, cancellationToken).ConfigureAwait(false);
+            }
+
+            return packageInfo;
+        }
+
+        private static async Task<NpmPackageInfo> GetPackageInfoForScopedPackageAsync(string packageName, CancellationToken cancellationToken)
+        {
+            Debug.Assert(packageName.StartsWith("@", StringComparison.Ordinal));
+            string searchName = "@" + HttpUtility.UrlEncode(packageName.Substring(1));
```

Comment:
Might be worth adding a comment here that we're trying to avoid encoding the @ sign.  At a glance it may seem weird to do the concatenation with `packageName.Substring(1)` until one realizes the actual intent.